### DISCUSSION TITLE
android: teamtalk does not use ipv6 network

### DISCUSSION THREAD

**Discussion by jeremyp3** [BearWare/TeamTalk5#1525] [Q&A]
### Description

android does not support ipv6 nnetwork

### Application

- [ ] qtTeamTalk
- [X] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkClassic
- [ ] TeamTalkServer

### Platform

- [ ] Windows
- [ ] macOS
- [X] Android
- [ ] iOS
- [ ] Linux

### Expected behavior

I see that teamtalk 5 on android does not support ipv6 connections. 
on iOS, it works.

is this an oversight?

thanks,

### Steps to reproduce problem

lunch teamtalk with ipv6 server

**Comment by bear101** [comment]
Looking at the [Android configuration](https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalkLib/build/ace/CMakeLists.txt#L470) it looks the same as iOS. Are you sure you've configured your IPv6 properties correctly?

**Comment by jeremyp3** [comment]
Yes. my ipv6 is well configured. moreover, I asked other people to try it too to see if they have the same problem, and they do. I forgot to mention it in my first message, sorry.

**Comment by bear101** [comment]
Do you connect to a hostname or an IP-address?

**Comment by jeremyp3** [comment]
I connect using a hostname. 
- if i use an ipv6 only hostname, it doesn't connect. it seems to say, unable to connect 
- If I use an ipv6 / ipv4 hostname, it connects as ipv4

I have not tested with a literal ipv6 address

**Comment by jeremyp3** [comment]
hi,

I confirm that I have the same problem using an ip address directly. The message is: the connection failed. 

my phone has IPV6 connectivity.

**Comment by Nardol** [comment]
I confirm this behavior, I cannot connect to an IPV6 only TeamTalk server with the same result as @jeremyp3 when using Android after checking IPV6 connectivity is OK of course.

**Comment by jeremyp3** [comment]
Are there news on the subject?
Teamtalk in IPv6 still does not work on Android, while my Android phone at IPv6

**Reply by bear101** [reply]
TeamTalk is compiled with IPv6 support on Android: https://github.com/BearWare/TeamTalk5/blob/master/Library/TeamTalkLib/build/ace/CMakeLists.txt#L459

This works on iOS and Linux, so I don't see why it would not be working on Android.

**Comment by jeremyp3** [comment]
@bear101

**Comment by jeremyp3** [comment]
@bear101 
in any case, today I confirm and several friends confirm that it does not work for them either.

we tested from android 10 to android 13 with several different brands of phones, and there is no ipv6, neither in wifi, nor in 4/5g. 
The phones do have ipv6 connectivity. 

we can surf the internet in ipv6, we can use other services in ipv6...

can we help you in any way if you don't have ipv6 to try? 

thanks,

Jeremyp3

**Comment by jeremyp3** [comment]
@bear101 
in any case, today I confirm and several friends confirm that it does not work for them either.

we tested from android 10 to android 13 with several different brands of phones, and there is no ipv6, neither in wifi, nor in 4/5g. 
The phones do have ipv6 connectivity. 

we can surf the internet in ipv6, we can use other services in ipv6...

can we help you in any way if you don't have ipv6 to try? 

thanks,

Jeremyp3

**Comment by lbk2907** [comment]
@bearware101
On my site, IPV6 also not working on TeamTalk for Android. I can confirm 
that the IPV6 server is running correctly.

**Comment by hejiadong0608** [comment]
@bear101 Yes, I confirmed that it cannot connect when connecting to the IPV6 server.

**Comment by lbk2907** [comment]
@bear101
Still can't connect here using IPv6 on TeamTalk5Droid. On Windows and iTeamTalk has no problem.
Here's my IPV6 just in case you want to take a look. I not have domain.
2400:6180:0:d1::552:1
Server Hosting Provider: Digital Ocean

**Comment by bear101** [comment]
I've submitted a new TeamTalk for Android (v5.15.2). Can you check if the IPv6 issue has been fixed by some miracle?

**Comment by lbk2907** [comment]
Do you submit an update at Play Store? I didn't see the update here yet. If
not, I'll compile by myself and give it a try.
